### PR TITLE
Sortby

### PR DESCRIPTION
**Pull request by TravelByBeaut**
Adds sortby, orderby and topic filter with error handling

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -124,6 +124,98 @@ describe("Articles", () => {
...
+        body: "some gifs",
+        created_at: "2020-11-03T09:12:00.000Z",
+        votes: 0,
+        comment_count: 2,
+      };
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body: { article } }) => {
+          expect(article[0]).toMatchObject(articleExample);
```
Comment (karlaferry) [inline __tests__/app.test.js]: Please do not test the queries this way, use jest-sorted instead!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -124,6 +124,98 @@ describe("Articles", () => {
...
+        });
+    });
+    test("status: 404 when given a non-existent topic", () => {
+      return request(app)
+        .get("/api/articles?topic=none")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Query does not exist");
+        });
+    });
```
Comment (karlaferry) [inline __tests__/app.test.js:183]: Please add a test for when a query is a non-existent query (ie. `/api/articles?notValid=nope`), this should not throw an error but should return with the default queries set.

**Review by karlaferry — CHANGES_REQUESTED** [review]
Please have a look at using jest-sorted, very important!

File: __tests__/app.test.js [hunk 3]
```diff
@@ -125,37 +125,35 @@ describe("Articles", () => {
         });
     });
     test("adds sort by functionality default sorts columns by date", () => {
```
Comment (Rosem10) [inline __tests__/app.test.js:128]: In the description you've said this should default to be sorted by date but this test doesn't actually test whether the responses are sorted by date.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -125,37 +125,35 @@ describe("Articles", () => {
...
-        body: "We all love Mitch and his wonderful, unique typing style. However, the volume of his typing has ALLEGEDLY burst another students eardrums, and they are now suing for damages",
-        created_at: "2020-05-06T01:14:00.000Z",
-        votes: 0,
-      };
       return request(app)
         .get("/api/articles?sort_by=author")
         .expect(200)
         .then(({ body: { article } }) => {
-          expect(article[0]).toMatchObject(articleExample);
+          expect(article).toHaveLength(12);
```
Comment (Rosem10) [inline __tests__/app.test.js]: Same as above, except now we're not testing if this defaults to descending.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -124,6 +124,102 @@ describe("Articles", () => {
...
+      return request(app)
+        .get("/api/articles?topic=none")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Query does not exist");
+        });
+    });
+    test("status: 200 and default queries when given a non-existent topic", () => {
+      return request(app)
+        .get("/api/articles?none=none")
```
Comment (Rosem10) [inline __tests__/app.test.js:186]: This should be topic = none

**Review by Rosem10 — CHANGES_REQUESTED** [review]
I'm not sure if this is something I missed, but I still don't think you're using jest-sorted. You should be able to assert that the response you're getting back toBeSortedBy and then the sortedBy value

File: __tests__/app.test.js (hunk 5 above)
Comment (TravelByBeaut) [reply __tests__/app.test.js:186]: It was meant to be where the query is non-existent

Comment (karlaferry) [reply __tests__/app.test.js:186]: If that's the case then test description should say non-existent query instead of non-existent topic.

Comment (TravelByBeaut) [reply __tests__/app.test.js:186]: I have changed it in the new upload :)

File: __tests__/app.test.js [hunk 6]
```diff
@@ -124,6 +125,70 @@ describe("Articles", () => {
...
+        .then(({ body }) => {
+          expect(body.msg).toBe("Query does not exist");
+        });
+    });
+    test("status: 200 and default queries when given a non-existent query", () => {
+      return request(app)
+        .get("/api/articles?none=none")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.article).toHaveLength(12);
```
Comment (karlaferry) [inline __tests__/app.test.js]: With this one you have to check using jest-sorted that it is actually sorted and ordered by the defaults

**Review by karlaferry — APPROVED** [review]
Approving but please add a quick jest sorted test for the bit where I left feedback on!

**Pull request merged**